test/common/stats/thread_local_store_test.cc

```diff
@@ -128,6 +128,7 @@ class HistogramTest : public testing::Test {
   }
 
   void TearDown() override {
+    tls_.shutdownGlobalThreading();
```

Comment:
what happens if we don't have these new calls to shutdownGlobalThreading? Will these unit tests crash or be flaky?

The concern is whether that flow would also occur in integration tests or in production.

Reply:
I will run again with those calls removed to be sure it'd pass. However, the lack of those calls being made is why I did not catch breaking other tests in my previous push. `server/server.cc` will call shutdownGlobalThreading, then shutdownThreading, then shutdownThread, so the update was to be more in line with how Envoy will run outside integration tests.

Will confirm though.

Reply:
Let's make sure we have both paths covered in unit tests then: calling shutdownGLobalThreading and also not calling it.

Reply:
Okay. I tried remove all the shutdownGlobalThreading calls I added and the integration test still passes now. I would expect others to as well. I'm not sure where we'd like to make the distinction of where to call shutdownGlobalThreading and where not to. We could do it as part of ThreadLocalStoreNoMocksTestBase (and individual tests that call shutdownThreading prior to the destructor).

It may be worth noting ThreadLocalRealThreadsTestBase already does call shutdownGlobalThreading in the destructor, so perhaps I should just remove the calls so both paths are hit.

Reply:
Sounds good.